### DISCUSSION TITLE
Idea: Redirect console warnings

### DISCUSSION THREAD

**Discussion by crishoj** [jisotalo/ads-client#151] [General]
`ads-client` has the option `hideConsoleWarnings` to suppress console warnings.

For applications with centralized logging, it could be useful to "catch" these warnings from `ads-client` rather than suppressing them. 

Perhaps an option for a custom console logging function:

```ts
{
  consoleLogger?: (message?: any, ...optionalParams: any[]) => void
}
```

It could default to `console.log`.

**Comment by jisotalo** [comment]
Actually I just thinking when I was writing the documentation that there should be a `console` event. So it would be possible to catch all console output using `client.on('console', handleConsole)` or similar.

What do you think?

**Reply by crishoj** [reply]
A `console` event would be perfect!

**Reply by crishoj** [reply]
Proposal (with event name `warning`): #152

**Comment by jisotalo** [comment]
@crishoj 

Probably related to #150 as well.

Should we change this unknown notification `client-error` to a `warning` event instead? That would suit the case better.
https://github.com/jisotalo/ads-client/blob/ed8826ac1a67fa521ffcb1ea245e3dfccca81e2d/src/ads-client.ts#L1833

**Reply by jisotalo** [reply]
Actually, it should be a `warn()` call instead? That would make more sense.

**Reply by crishoj** [reply]
Agreed, that makes sense

**Reply by crishoj** [reply]
> Probably related to https://github.com/jisotalo/ads-client/discussions/150 as well.

I could include the proposed optional automatic deletion of stale notification handles in PR #152, or do a separate PR with that.

**Reply by jisotalo** [reply]
Thanks! A separate PR sounds good
